src/test_l3_plat_power.c

```diff
@@ -0,0 +1,469 @@
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
+{
+   int c;
+   while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+* @brief Initialization of the HAL Power Manager Module
+*
+* This test provides a scope to open the HAL Power Manager module.
+
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 001@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_l3_power_manager_hal_Init(void)
+{
+    gTestID = 1;
+    pmStatus_t status = PWRMGR_SUCCESS ;
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    // Step 1: Call PLAT_INIT()
+    UT_LOG_INFO("Calling PLAT_INIT()");
+    status = PLAT_INIT();
+    UT_LOG_INFO("Result PLAT_INIT: pmStatus_t:[%s]",
+                 UT_Control_GetMapString(pmStatus_mapTable, status));
+
+    UT_ASSERT_EQUAL_FATAL(status, PWRMGR_SUCCESS);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Use case of set power state of the HAL Power Manager Module
+*
+* This test provides a scope to set the power state
+
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 002@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_power_manager_hal_Set_Power_State(void)
+{
+    gTestID = 2;
+    pmStatus_t status = PWRMGR_SUCCESS ;
+
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    int32_t  powerState =0;
+
+     //Step 1: Get the Power State
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Supported Power States");
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("\t#  %-20s","Power State");
+    for(int32_t i = PWRMGR_POWERSTATE_OFF; i <= PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP; i++)
+    {
+        UT_LOG_MENU_INFO("\t%d.  %-20s", i, UT_Control_GetMapString(plat_power_states_mapTable, i));
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Select Power State: ");
+    scanf("%d", &powerState);
+    readAndDiscardRestOfLine(stdin);
+    if(powerState < PWRMGR_POWERSTATE_OFF || powerState > PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP)
+    {
+        UT_LOG_ERROR("Invalid Power State");
+        goto exit;
+    }
+    PWRMgr_PowerState_t state = (PWRMgr_PowerState_t)powerState;
+
+    // Step 2: Call PLAT_API_SetPowerState()
+    UT_LOG_INFO("Calling PLAT_API_SetPowerState()");
+    status = PLAT_API_SetPowerState( state );
+    UT_LOG_INFO("Result PLAT_API_SetPowerState: pmStatus_t:[%s]",
+                 UT_Control_GetMapString(pmStatus_mapTable, status));
+
+    UT_ASSERT_EQUAL_FATAL(status, PWRMGR_SUCCESS);
```

Comment:
use `assert` so that test will exit app rather than current test case
please check this 
https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/develop/src/test_l3_dsVideoPort.c#L327
https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/develop/src/test_l3_dsAudio.c#L275

Reply:
Updated to use assert.